jax/_src/tree.py

```diff
@@ -293,7 +293,7 @@ def field(*, static: bool = False, metadata: dict[str, Any] | None = None, **kwd
 
   See {func}`jax.tree.dataclass` for details.
   """
-  return dataclasses.field(metadata=(metadata or {}) | {'static': static}, **kwds)
+  return dataclasses.field(metadata={**(metadata or {}), 'static': static}, **kwds)
```

Comment:
I don't know if this is on purpose, but this does not prevent `metadata` from having a `static` key.

```python
In [1]: {**{"static": True}, "static": False}
Out[1]: {'static': False}

In [2]: dict(**{"static": True}, static=False)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[2], line 1
----> 1 dict(**{"static": True}, static=False)

TypeError: dict() got multiple values for keyword argument 'static'
```

Reply:
We'll figure out these details before merging – we're a long way from that.